File: ransomnote/ransomnote.py
def can_construct1(ransomNote, magazine):
    if len(magazine) < len(ransomNote):
        return False

    magKey = {}
    for char in magazine:
        if magKey.get(char):
            magKey[char]+=1
        else:
            magKey[char] = 1

    for char in ransomNote:
        if not magKey.get(char) or magKey[char] == 0:
            return False
        else:
            magKey[char]-=1

    return True

File: ransomnote/ransomnote.js
function canConstruct1(ransomNote, magazine) {
    if (magazine.length < ransomNote.length) {
        return false
    }

    const magKey = {}
	for (let char of magazine) {
		if (magKey[char]) {
			magKey[char]++
		} else {
			magKey[char] = 1
		}
    }

    for (let char of ransomNote) {
        if (typeof magKey[char] === 'undefined' || magKey[char] === 0) {
            return false
        } else {
            magKey[char]--
        }
    }

    return true
}
